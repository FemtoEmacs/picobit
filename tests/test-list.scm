;; tests with lists

(define lst1 '())
(define lst2 '(1 2 3 4))
(define lst3 '(a b c d))
(define lst4 '(#\a #\b #\c #\d))

(display (null? lst1))
(display (not (null? lst2)))
(display (length lst1))
(display (length lst2))
(display (append lst2 lst3))
(display (reverse lst4))

(display (list->string lst4))
(display (string->list "abcd"))

(list-set! lst2 2 5)
(display (list-ref lst2 2))
