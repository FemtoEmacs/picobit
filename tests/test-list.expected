:0080000080
:10800000FBD73204800501328065012F81284000B2
:10801000812C0114813440008119400081134000FB
:1080200081104000811240008066012180070122FA
:10803000804200028027012381240002802701253D
:10804000802401268127011D807101298067012A72
:10805000806E012B806800028027012D8077012E21
:10806000807301308024000281310111806A000296
:108070008073013380670135806701188008000232
:108080008040011A800020008040011B8032011CCA
:10809000800020008027011E8066011F806800028A
:1080A000806901208000200080780002807D011519
:1080B0008066011680660117800020008006010E90
:1080C0008070010F8065010D8078010F02B4024EAF
:1080D0005202B4023F5102B4022D5002B40227539F
:1080E00002D2B5C5EEA104D2D4B5BEEE02B00215DF
:1080F000B5B7EEA104B00215B5AFEEA104A107B06B
:108100000206B5A5EEA105B001F3B59DEEA105DC13
:10811000B597EEA108DBB591EEA10406094063EE28
:10812000A104064162B582EED80120DAB895432158
:10813000DBB6800221CC9A21CE2161EE2022CFB67F
:108140007200EF20B6800120DA9EA02607E620B557
:1081500058EEA02607E600EF20C0B8BF20B582B6D3
:1081600048012004C8B8AEA03121B5A402B583CD22
:10817000DCEF02A034210EC5C122CD2102CDB40115
:10818000827101210EC89220EF210E426221B66257
:10819000010421C2EF2002B559B75520CCB8B8A0D0
:1081A0002C07E620CEB51FEE20CFB6800120D29658
:1081B000A02D07E600EF20CC9DA02407E620CEB539
:1081C00005EE20CFB666A10AB00129EE20B0014627
:1081D000EEA02D07E600EF20D995A10BB1012920D3
:1081E000F09C2095A109B10129A106B10129A10C9A
:1081F000B10129012002B6800220CC9820CF21CEE7
:1082000023CDB67421EF0220CC9920CE21CF23B507
:1082100075CDEF21EF012004B6800220CC9720CF4E
:108220002205C1B67521EF012007E600EF03210406
:10823000C7952023D000EF20CF2205C2244073022F
:108240002104C79320CEEF20CF2205C24172022025
:1082500004C8212302CDCDB4025B71012204C822DF
:10826000242302CDCDCDB4026A710123B8DC21B044
:10827000019023222602CDCDCDB4027D710122B81A
:10828000C621B0019021242602CDCDCDB40290713B
:10829000012022242602CDCDCDCDB4029E70002136
:1082A00021C4232502CDCDB402AB710122B89321A4
:1082B0009220EF2298019420B1019020EF21B87B09
:0D82C000B77500B86EB76A21B73B21B7252E
:00000001FF
